taichi/aot/graph_data.h

```diff
@@ -23,10 +24,38 @@ struct Arg {
   ArgKind tag;
   std::string name;
   // TODO: real element dtype = dtype + element_shape
-  std::string dtype_name;
+  PrimitiveTypeID dtype_id;
   std::vector<int> element_shape;
 
-  TI_IO_DEF(name, dtype_name, tag, element_shape);
+  // For serialization & deserialization
+  explicit Arg()
+      : tag(ArgKind::kUnknown),
+        name(""),
+        dtype_id(PrimitiveTypeID::unknown),
+        element_shape({}) {
+  }
+
+  explicit Arg(ArgKind tag,
+               std::string name,
```

Comment:
nit: for all PRs, prefer `const std::string&` (unless you can make sure the callers uses `std::move(str)`).